projects/vdk-plugins/vdk-properties-fs/tests/test_fs_properties_plugin.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import os
+import shutil
+import tempfile
+import uuid
+from typing import Dict
+from unittest.mock import patch
+
+from click.testing import Result
+from vdk.plugin.properties_fs import fs_properties_plugin
+from vdk.plugin.properties_fs.fs_properties_client import (
+    FileSystemPropertiesServiceClient,
+)
+from vdk.plugin.test_utils.util_funcs import cli_assert_equal
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_funcs import jobs_path_from_caller_directory
+
+FS_PROPERTIES_DIRECTORY = "FS_PROPERTIES_DIRECTORY"
+FS_PROPERTIES_FILENAME = "FS_PROPERTIES_FILENAME"
+PROPERTIES_DEFAULT_TYPE = "PROPERTIES_DEFAULT_TYPE"
+
+# custom directory and filename generated
+directory = os.path.join(tempfile.gettempdir(), str(uuid.uuid4()))
+filename = str(uuid.uuid4())
+
+
+@patch.dict(
+    os.environ,
+    {
+        PROPERTIES_DEFAULT_TYPE: "fs-properties-client",
+    },
+)
+def test_multiple_properties_read_write_config_default():
+    runner = CliEntryBasedTestRunner(fs_properties_plugin)
+    # default file path expected
+    file_path = os.path.join(tempfile.gettempdir(), "vdk_data_jobs.json")
```

Comment:
tempfile doesn't work very well since it does not actually delete the files ,. It simply places them in temp directory.

For python tests it's recommended to use tmp path fixture - https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html  

Probably something like this: 
```suggestion
def test_multiple_properties_read_write_config_default(tmp_path):
    runner = CliEntryBasedTestRunner(fs_properties_plugin)
    # default file path expected
    file_path = os.path.join(str(tmp_path), "vdk_data_jobs.json")
```

Reply:
I'm finalising the test file resources (except in case of reusing the default file path); will be merging since we need now the plugin released for demo prep